### PR TITLE
Cookbook Updates

### PR DESCRIPTION
**Pull request by bramwelt**
This PR brings the `code_generator` cookbook closer in-line with our cookbook standards. It includes updates to:

- Berksfile
- .gitignore
- .rubocop.yml
- README.md
- serverspec.rb
- ... and more

It also includes foodcritic and rubocop enforcements.

### CODE REVIEW

File: recipes/cookbook.rb [hunk 1]
```diff
@@ -25,8 +25,29 @@
...
+end
+
+template "#{test_dir}/default_spec.rb" do
+    source "serverspec.erb"
+    helpers(ChefDK::Generator::TemplateHelper)
+    action :create_if_missing
+end
+
+
+
```
Comment (fahlmant) [inline recipes/cookbook.rb]: One too many newlines?

**Comment by jordane** [comment]
@bramwelt this branch generates rubocop errors for me:

```
Inspecting 11 files
..C........

Offenses:

recipes/cookbook.rb:42:1: C: Use 2 (not 4) spaces for indentation.
    source "serverspec.erb"
^^^^
recipes/cookbook.rb:42:12: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    source "serverspec.erb"
           ^^^^^^^^^^^^^^^^
recipes/cookbook.rb:47:1: C: Extra blank line detected.
recipes/cookbook.rb:48:1: C: Extra blank line detected.
```

File: recipes/cookbook.rb (hunk 1 above)
Comment (jordane) [inline recipes/cookbook.rb]: Two too many, actually.

File: templates/default/kitchen.cloud.yml.erb [hunk 2]
```diff
@@ -23,11 +26,11 @@ provisioner:
...
 
 platforms:
-- name: centos-6.5
+- name: centos-6.6
   driver_plugin: openstack
   driver_config:
     username: centos
-    image_ref: "CentOS 6.5"
+    image_ref: "CentOS 6.6"
 - name: ubuntu-12.04
```
Comment (ramereth) [inline templates/default/kitchen.cloud.yml.erb]: We should remove Ubuntu and replace it with CentOS 7.1.

File: templates/default/kitchen.cloud.yml.erb [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+  openstack_api_key: <%%= ENV['OS_PASSWORD'] %>
+  openstack_auth_url: <%%= "#{ENV['OS_AUTH_URL']}/tokens" %>
+  key_name: <%%= ENV['OS_SSH_KEYPAIR'] %>
+  private_key_path: <%%= ENV['OS_PRIVATE_SSH_KEY'] %>
+  public_key_path: <%%= ENV['OS_PUBLIC_SSH_KEY'] %>
+  openstack_tenant: <%%= ENV['OS_TENANT_NAME'] %>
+  floating_ip_pool: <%%= ENV['OS_FLOATING_IP_POOL'] %>
+  flavor_ref: <%%= ENV['OS_FLAVOR_REF'] %>
+  # Only use the following if you need to open up firewall ports
+  security_groups:
```
Comment (ramereth) [inline templates/default/kitchen.cloud.yml.erb]: This should be removed from the default template. Please look at the docs to align it with our current standards.

File: templates/default/serverspec.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'serverspec'
+
+set :backend, :exec
+
+describe file('/') do
```
Comment (ramereth) [inline templates/default/serverspec.erb]: Should we actually have a default test in here or does it fail if there isn't one?

File: templates/default/kitchen.cloud.yml.erb [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
-  openstack_username: <\%= ENV['OS_USERNAME'] %>
-  openstack_api_key: <\%= ENV['OS_PASSWORD'] %>
-  openstack_auth_url: <\%= "#{ENV['OS_AUTH_URL']}/tokens" %>
-  key_name: <\%= ENV['OS_SSH_KEYPAIR'] %>
-  private_key_path: <\%= ENV['OS_PRIVATE_SSH_KEY'] %>
-  public_key_path: <\%= ENV['OS_PUBLIC_SSH_KEY'] %>
-  openstack_tenant: <\%= ENV['OS_TENANT_NAME'] %>
-  floating_ip_pool: <\%= ENV['OS_FLOATING_IP_POOL'] %>
-  flavor_ref: <\%= ENV['OS_FLAVOR_REF'] %>
+  openstack_username: <%%= ENV['OS_USERNAME'] %>
```
Comment (jordane) [inline templates/default/kitchen.cloud.yml.erb]: Yo dawg, I hear you like templates. but why are there now two `%`?

Comment (bramwelt) [inline templates/default/kitchen.cloud.yml.erb]:
```
Yo dawg, I heard you like erb templates, so we put an erb template 
in your erb template, so you can escape your variables 
while escaping your variables.
```

`%%` escapes the `%`, not `\%`

File: templates/default/serverspec.erb (hunk 4 above)
Comment (bramwelt) [inline templates/default/serverspec.erb]: There probably doesn't need to be a test in here. I'll check if test kitchen complains without tests, and remove it if it doesn't matter.

File: templates/default/kitchen.cloud.yml.erb (hunk 3 above)
Comment (bramwelt) [inline templates/default/kitchen.cloud.yml.erb]: You're specifically referring to `security_groups`?

File: templates/default/kitchen.cloud.yml.erb (hunk 2 above)
Comment (bramwelt) [inline templates/default/kitchen.cloud.yml.erb]: I don't believe we run any Ubuntu, so this change makes a lot of sense. For cookbook we would like to open source and support multiple platforms we can manually add them in.

File: templates/default/kitchen.cloud.yml.erb (hunk 5 above)
Comment (jordane) [inline templates/default/kitchen.cloud.yml.erb]: Oh, good to know.

File: templates/default/kitchen.cloud.yml.erb (hunk 3 above)
Comment (jordane) [inline templates/default/kitchen.cloud.yml.erb]: Yes.

**Comment by bramwelt** [comment]
:guardsman:
:small_red_triangle_down:

**Comment by bramwelt** [comment]
I have:
- Added a new cookbook README
- Added a CHANGELOG
- Fixed the rubocop issues

@osuosl-cookbooks/chefs Ready for review.

**Comment by jordane** [comment]
@bramwelt this is still failing a foodcritic issue:

```
foodcritic .
FC005: Avoid repetition of resource declarations: ./recipes/cookbook.rb:8
```

**Comment by bramwelt** [comment]
@jordane FC005 should be resolved.

**Comment by jordane** [comment]
@bramwelt FC005 is fixed, but you introduced a Ruby Syntax error!

```
ruby -c recipes/cookbook.rb
recipes/cookbook.rb:7: syntax error, unexpected keyword_do, expecting end-of-input
%w(metadata.rb README.md CHANGELOG.md) do |f|
```

**Comment by bramwelt** [comment]
@jordane Nice catch. Fixed and changes pushed.

**Comment by jordane** [comment]
+1

**Pull request merged**